libraries/AP_Compass/AP_Compass_MAG3110.cpp

```diff
@@ -251,7 +255,7 @@ void AP_Compass_MAG3110::read()
         return;
     }
 
-    Vector3f field(_mag_x_accum, _mag_y_accum, _mag_z_accum);
```

Comment:
did you find this change made a difference? It should be equivalent (and the original code should be very slightly faster)
If this did change behavior then we need to fix that elsewhere

Reply:
yes I think it too but first  code leaves garbage in 'field' variable somehow, checked by debugger. 2nd variant compiles into pure calculations without any memory copy/move.

Reply:
this may be not GCC issue but seems that Eclipse bug, but I want to see variables